### ISSUE TITLE
Image push to registry fails when using `--remote_download_minimal` and lazy push strategy

### ISSUE THREAD

**Issue by chardin-spacex** [tweag/rules_img#384]
Running an `image_push` bazel command with `--remote_download_minimal` fails with a misleading auth error:

`bazel run //<redacted>:image_push --remote_download_outputs=minimal`

> Error during deploy: deploying images: Patch "https://<redacted>/blobs/uploads/062e6e91-0491-425c-aa73-253aacf2715a": from Bazel remote cache: rpc error: code = Unauthenticated desc = User not found

The issue appears to be that the `*.tar.zst` build artifacts that are produced by the corresponding `image_layer` rule are not properly declared as transitive dependencies of the `image_push` rule

I can resolve this issue by including a flag that force downloads these artifacts from the remote server

`bazel run //<redacted>:image_push --remote_download_outputs=minimal --remote_download_regex=".*tar\.zst"`

The following config is also used present in my `.bazelrc`
```
# Ensure images layers are always compressed
common --@rules_img//img/settings:compress=zstd
common --@rules_img//img/settings:push_strategy=lazy
common --@rules_img//img/settings:remote_cache=grpcs://<redacted>
```

**Comment by malt3** [comment]
The error message contains a key detail: `rpc error: code = Unauthenticated desc = User not found`. When using the lazy push strategy, rules_img needs to connect (and authenticate) to your Bazel remote cache.
It looks like the cache under `grpcs://<redacted>` requires some form of authentication that is not provided to rules_img.

It is recommended that you configure a credential helper for rules_img or find a way to connect to the remote cache without requiring authentication.

Here's the way to configure the credential helper (either via an environment variable or a flag):

```
# Credential helper to use for authenticating gRPC connections during push operations
# in some push strategies.
# This can be the same as Bazel's credential helper.
# Falls back to $IMG_CREDENTIAL_HELPER env var.
common --@rules_img//img/settings:credential_helper=tweag-credential-helper
```

I recommend [tweag-credential-helper](https://github.com/tweag/credential-helper) for this purpose ([doc](https://github.com/tweag/credential-helper/blob/main/docs/providers/remoteapis.md)), but you can also write your own, as long as it implements the [credential helper spec](https://github.com/EngFlow/credential-helper-spec).

**Comment by malt3** [comment]
Closing since I believe this is not a bug in rules_img. Feel free to reopen with more details.
